components/org.wso2.carbon.config/src/main/java/org/wso2/carbon/config/provider/ConfigProvider.java

```diff
@@ -48,4 +48,19 @@
      * @throws ConfigurationException if there is a problem while reading the configurations
      */
     Object getConfigurationObject(String namespace) throws ConfigurationException;
+
+    /**
+     * Returns configuration object of the class.
+     * If configuration namespace is given, reads the configuration of the namespace and create instance of the the
+     * class from that configurations.
+     * If namespace is null or configuration doesn't exist for in configuration file, returns configurations based on
+     * the bean class.
+     *
+     * @param namespace config namespace
+     * @param configClass configuration bean class
+     * @param <T>        object type
+     * @return configuration object
+     * @throws ConfigurationException if there is a problem while reading the configurations
+     */
+    <T extends Object> T getConfigurationObject(String namespace, Class<T> configClass) throws ConfigurationException;
```

Comment:
Is it required to have `extends Object` here. AFAIK it is implied and hence we do not need to specify. WDYT?